packages/experiments/src/components/Pagination/Pagination.base.tsx

```diff
@@ -0,0 +1,148 @@
+
+export class PaginationBase extends BaseComponent<IPaginationProps, {}> {
+  public static defaultProps: Partial<IPaginationProps> = {
+    nextLabel: '>>',
+    previousLabel: '<<',
+    selectedPageIndex: 0,
+    pageRange: 2,
+    marginPagesDisplayed: 1,
+    omissionLabel: '...'
+  };
+
+  private _classNames: { [key in keyof IPaginationStyles]: string };
+
+  constructor(props: IPaginationProps) {
+    super(props);
+  }
+
+  public render(): JSX.Element {
+    const { nextAriaLabel, nextLabel, pageCount, previousAriaLabel, previousLabel, selectedPageIndex, styles, theme } = this.props;
+
+    this._classNames = getClassNames(styles!, {
+      theme: theme!
+    });
+
+    const canPrevious = selectedPageIndex! > 0;
+    const canNext = selectedPageIndex! + 1 < pageCount;
+
+    // FocusZone handles A11Y requirement such as:
+    // Dynamically set tabindex (0 or -1) to the correct item
+    // Refocus to the active item when component re-renders
+    // Handles all key strokes, e.g., left/right, space, enter
+    return (
+      <FocusZone direction={FocusZoneDirection.horizontal}>
+        <ul className={this._classNames.root} role="tablist">
+          <li key="previousPage">
+            <button
+              className={this._classNames.previousNextPage}
+              onClick={this.handlePreviousPage}
+              disabled={!canPrevious}
+              role="tab"
+              aria-label={previousAriaLabel}
+            >
+              {previousLabel}
+            </button>
+          </li>
+          {this._pageList()}
+          <li key="nextPage">
+            <button
+              className={this._classNames.previousNextPage}
+              onClick={this.handleNextPage}
+              disabled={!canNext}
+              role="tab"
+              aria-label={nextAriaLabel}
+            >
+              {nextLabel}
+            </button>
+          </li>
+        </ul>
+      </FocusZone>
+    );
+  }
+
+  private handleSelectedPage = (selected: number) => {
+    if (selected === this.props.selectedPageIndex) {
+      return; // same page, no action
+    }
+    if (this.props.onPageChange) {
+      this.props.onPageChange(selected);
+    }
+  };
+
+  private handlePreviousPage = () => {
+    this.handleSelectedPage(this.props.selectedPageIndex! - 1);
+  };
+
+  private handleNextPage = () => {
+    this.handleSelectedPage(this.props.selectedPageIndex! + 1);
+  };
+
+  private _pageElement(index: number): JSX.Element {
+    return (
+      <PageNumber
+        key={index + 1}
+        page={index + 1}
+        pageAriaLabel={this.props.pageAriaLabel}
+        selected={index === this.props.selectedPageIndex}
+        applyPage={this.handleSelectedPage}
+        className={this._classNames.pageNumber}
+      />
+    );
+  }
+
+  private _pageList(): JSX.Element[] {
+    const pageList = [];
+    if (this.props.pageCount <= this.props.pageRange!) {
```

Comment:
Readability. Make use of object destructuring on `this.props`. Thanks!


![screen shot 2018-10-03 at 4 30 28 pm](https://user-images.githubusercontent.com/29514833/46445141-1e8e5b80-c72a-11e8-80b0-dcb56ba7b158.png)

Reply:
Thank you for highlighting these. I updated these in Iteration 9. 

---
In reply to: [222498672](https://github.com/OfficeDev/office-ui-fabric-react/pull/6532#discussion_r222498672) [](ancestors = 222498672)